### DISCUSSION TITLE
Insert / update multiple records - JSON format

### DISCUSSION THREAD

**Discussion by dmccallie** [pocketbase/pocketbase#1594] [Q&A] [answered]
This is a pretty basic question, but I couldn't find an answer. So here goes.

I have a simple "tags" collection that tracks user-supplied hashtags to a blogging system.
The "tags" collection is super simple to start with - just a text field for the tag string and a relation field with the blogId (reference to the blogs collection.

I want to have each tag be a separate record (perhaps a bad idea, but that's what I am trying so far...)  

How would I insert a set of new tags for an existing blogId? 

1) What should the JSON look like for the initial create call? This is what I tried but didn't work. I may have other errors of course...

```
data = [
    {
        'tag' : '#hashtext_1",
        'blogId: '123456789'
    },
    {
        'tag' : '#hashtext_2",
        'blogId: '123456789'
   },
];
...
pb.collection('tags').create( data );
```

Thanks for putting up with such basic questions!

**Comment by dmccallie** [comment, marked as answer]
OK, I think I found the answer - apparently creating multiple records will require multiple independent calls to

`pb.collection("tags").create( data )`

I belatedly found an example in question #853.

Something like:

```
function createRecordsInParallel(pb, collectionName, records) {
	const promises = records.map((record) => {
		return pb.collection(collectionName).create(record, {
			'$autoCancel': false,
		});
	})
	return Promise.all(promises);
}
```

Is there any plan to support a multi-row insert via a single call? 
(Thanks again for a very nice project)

**Reply by ganigeorgiev** [reply]
> Is there any plan to support a multi-row insert via a single call?

Bulk and transactional create/update/delete is in the roadmap and eventually will be handled as part of #48.
